### PR TITLE
perf(app): move content layer reads to server side

### PR DESCRIPTION
**Pull request by H4ad**
## Summary

Move content layer reads to server-side, reducing from 130mb to 10mb of resource fetch on /guides page.

## Motivation / Problem

The current code was importing all blogs/guides/everything into client side.

/guides page loaded 90mb of JS file due do this.

**Important Notes**: Even on RSC, if we pass the whole allBlogs, it still ends up in the user bundle when the user makes a request via `?rsc`, likely on prefetch. To avoid the damages, I also added a map to exclude body and avoid at all costs having the entire blog post to be embedded.

## Changes

Move all content layer reads to server side only, and pass it as props to the components reading it.

## Description

To reduce bundle size.

## Type of Change

- [ ] **Docs** – Changes to `data/docs/**`
- [ ] **Blog** – Changes to `data/blog/**`
- [x] **Site Code** – Changes to `app/**`, `components/**`, `hooks/**`, `utils/**`, config, etc.
- [ ] **Redirects** – Renamed/moved docs or updated `next.config.js` redirects
- [ ] **Dependencies / CI / Scripts**
- [ ] **Other** – Explain below

## Impact

- [ ] Documentation only
- [ ] UI changes
- [ ] Developer workflow
- [x] Performance impact
- [ ] Breaking change

## Before / After (if applicable)

Before:

image

After:

image

### General
- [x] Branch is up to date with `main`
- [x] PR title follows conventional format (`type: description`)
- [x] Self-reviewed the code
- [x] Comments added for complex logic

### For all changes
- [x] Built locally (`yarn build`) with no errors
- [x] Ran `yarn lint` and fixed any issues
- [x] Pre-commit hooks passed (or ran `yarn check:doc-redirects` / `yarn check:docs-metadata` if applicable)

### For docs changes (`data/docs/**`)
- [ ] Followed the docs author checklist in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added/updated the page in `constants/docsSideNav.ts` if adding or moving a doc
- [ ] Any added docs images use WebP format and live under `public/img/docs/<topic>/`

### For blog changes
- [ ] Followed the blog workflow in [contributing/blog-workflow.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/blog-workflow.md)
- [ ] Frontmatter includes `title`, `date`, `author`, `tags` (and `canonicalUrl` if applicable)
- [ ] Images use WebP format and live under `public/img/blog/<YYYY-MM>/`

### For site code changes
- [ ] Followed the site code playbook in [contributing/site-code.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/site-code.md)
- [ ] New dependencies are justified in the PR description (if any)

### For renamed or moved docs
- [ ] Followed the redirects and discovery section in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added permanent redirect in `next.config.js` under `async redirects()`
- [ ] Updated internal links and sidebar in `constants/docsSideNav.ts`
- [ ] Ran `yarn check:doc-redirects` to verify

## Testing

- [x] Tested locally
- [x] Edge cases considered
- [x] Existing functionality verified
- [x] New tests added (if applicable)

Steps to test:
1. Open Guides
2. Press F12
3. Refresh the page
4. See network tab

### CODE REVIEW

File: app/resource-center/opentelemetry/page.tsx [hunk 1]
```diff
@@ -36,20 +43,27 @@ export const metadata: Metadata = {
...
   } catch (error) {
     console.error('Error fetching OpenTelemetry articles:', error)
   }
 
-  return <OpenTelemetryClient initialArticles={articles} comparisonPosts={comparisonPosts} />
+  return (
+    <OpenTelemetryClient
+      initialArticles={articles}
+      comparisonPosts={comparisonPosts}
+      blogPosts={getResourceCenterBlogs()}
```
Comment (manika-signoz) [inline app/resource-center/opentelemetry/page.tsx]: function call output should be stored in a constant and then passed to component similar to https://github.com/SigNoz/signoz.io/blob/7112cddbd6ad474156e55b5f563d3ca205fb1a1a/app/resource-center/guides/page.tsx#L6 pattern in the PR

File: app/resource-center/page.tsx [hunk 2]
```diff
@@ -1,6 +1,12 @@
 import React from 'react'
 import ResourceCenterClient from './ResourceCenterClient'
+import { getResourceCenterBlogs, getResourceCenterGuides } from './content'
 
 export default async function ResourceCenter() {
-  return <ResourceCenterClient />
+  return (
+    <ResourceCenterClient
+      blogPosts={getResourceCenterBlogs()}
```
Comment (manika-signoz) [inline app/resource-center/page.tsx]: similar to https://github.com/SigNoz/signoz.io/pull/3056/changes#r3055746690

File: app/resource-center/opentelemetry/page.tsx (hunk 1 above)
Comment (H4ad) [reply app/resource-center/opentelemetry/page.tsx]: Fixed

File: app/resource-center/page.tsx (hunk 2 above)
Comment (H4ad) [reply app/resource-center/page.tsx]: Fixed

**Review by manika-signoz — APPROVED** [review]

**Review by YounixM — APPROVED** [review]
